### PR TITLE
Implemented changes for shallow behavior in EIB

### PR DESCRIPTION
**Pull request by iamrishavkumar**
### Applicable Issues
Implemented changes for shallow behavior in EIB

### Description of the Change
Providing shallow parameter as true by default in EI-Backend , will limit the ER query from EIB to search for upstream/downstream events in main ER only. Search will not be navigated to external ER(s). Configurable parameter **event.repository.shallow** added in application.properties to support these changes.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Rishav-kumar rishav.kumar3@tcs.com

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -58,6 +58,7 @@ spring.mail.properties.mail.smtp.auth: false
 spring.mail.properties.mail.smtp.starttls.enable: false
 
 event.repository.url:
+event.repository.shallow:
```
Comment (tobiasake) [inline src/main/resources/application.properties:61]: You can set false value as default to shallow flag.
And also describe about what shallow flag is in configuraiton.md documentation file.

File: src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java [hunk 2]
```diff
@@ -100,14 +104,22 @@ private ResponseEntity sendRequestToER(String eventId, SearchOption searchOption
 
     private void prepareRequest(String eventId, SearchOption searchOption, int limit,
             int levels, boolean tree) throws IOException, URISyntaxException {
+        String shallow="";
         final SearchParameters searchParameters = getSearchParameters(searchOption);
+        if (shallowParameter == null || shallowParameter.isEmpty()) {
+            shallow ="true";
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java:110]: Has it been discussed what default value to use for shallow flag.
I guess this need to be documented in Markdown documentation.

**Review by tobiasake — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 1 above)
Comment (iamrishavkumar) [reply src/main/resources/application.properties:61]: OK , I will make the shallow flag as true and will add description about shallow flag in documentation

**Review by tobiasake — APPROVED** [review]
Looks good.

But for me it doesn't matter what default value we have to the shallow flag.
What default value will it be set to in RemRem?
We can use same default value in EI also.

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -58,6 +58,7 @@ spring.mail.properties.mail.smtp.auth: false
 spring.mail.properties.mail.smtp.starttls.enable: false
 
 event.repository.url:
+event.repository.shallow:true
```
Comment (tobiasake) [inline src/main/resources/application.properties:61]: Add a white space between the colon and "true" value, so it follows the the same format as other lines in this files.

**Review by tobiasake — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 3 above)
Comment (iamrishavkumar) [reply src/main/resources/application.properties:61]: Sure . I will do that

**Review by tobiasake — APPROVED** [review]

File: src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java [hunk 4]
```diff
@@ -53,6 +53,10 @@
     @Value("${event.repository.url}")
     private String eventRepositoryUrl;
 
+    @Getter
+    @Value("${event.repository.shallow}")
+    private String shallowParameter;
```
Comment (Surya-Kolli) [inline src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java:58]: cant you reasd this as boolean

Comment (iamrishavkumar) [reply src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java:58]: This shallowParameter value is required as string while creating http request to ER . Thats why kept it as string.

**Review by zburswa — APPROVED** [review]

**Review by Surya-Kolli — APPROVED** [review]

**Pull request merged**